### DISCUSSION TITLE
How to deploy the /public folder

### DISCUSSION THREAD

**Discussion by vladyslavnes** [community/community#21853] [Other Feature Feedback, Questions, & Ideas] [answered]
<p>I have a project with a two-folder structure: <a href="https://github.com/vladyslavnes/material-todo" rel="nofollow noopener">https://github.com/vladyslavnes/material-todo</a>. How do I deploy if index.html is in /public/ folder? Seems like GithHub Pages don’t understand me.</p>

**Comment by nadiajoyce** [comment]
<p>Hi <a class="mention" href="/vladyslavnes">@vladyslavnes</a>,</p>
<p>Thanks for being part of the GitHub Community Forum!</p>
<p>Your post was edited because it contained potentially sensitive information that you may not have intended to make public, such as your email address. We discourage users from sharing such information in our public forum, as it can create security and privacy risks for the poster, as well as other users.</p>
<p>Thanks so much!</p>

**Comment by birdcar** [comment]
<blockquote>
<p><span class="mention">@vladyslavneswrote</span>:</p>
</blockquote>
<blockquote>
<p>I have a project with a two-folder structure: <a href="https://github.com/vladyslavnes/material-todo" rel="nofollow noopener">https://github.com/vladyslavnes/material-todo</a>. How do I deploy if index.html is in /public/ folder? Seems like GithHub Pages don’t understand me</p>
</blockquote>
<p>GitHub Pages isn’t currently able to publish sites from any custom subdirectory, so it can’t be configured to make the “/public” directory the location of your site. However, if you place your site in a directory named “/docs”, then you can configure GitHub Pages to publish your site from there. More information can be found here:</p>
<p><a href="https://help.github.com/articles/configuring-a-publishing-source-for-github-pages/" rel="nofollow noopener">https://help.github.com/articles/configuring-a-publishing-source-for-github-pages/</a></p>

**Reply by birdcar** [reply]
> The limitation doesn't exist any more, you can set up your own publishing workflow: https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#publishing-with-a-custom-github-actions-workflow
>
> -- [@airtower-luna](https://github.com/orgs/community/discussions/21853#discussioncomment-4162048) 

While my response (currently marked as the "answer" here) _was_ correct at the time (2018), this is the current correct answer (as of Nov 22nd, 2022).

I tried to figure out how to de-mark my answer and update it to yours @airtower-luna, but unfortunately it looks like that's not possible for me to do.

**Reply by soundOfff** [reply]
I sum up to this answer since it's the correct one. ``public`` folder is not allowed even if you use a custom workflow

**Comment by manu-4216** [comment]
<p>This limitation doesn’t make any sense.</p>
<p>Please add the ability to serve gh-page from any folder, such as /public.</p>
<p>Since now you have the Microsoft support, and already adding other arguably less essential features, this paricular feature is a no-brainer, and very useful for many users.</p>

**Comment by AndreaGriffiths11** [comment]
<p>Thanks for the feedback! I’ve taken your suggestion and passed it along to the appropriate teams. Thanks again for reaching out :grinning:</p>

**Comment by hackerkid** [comment]
<p>Hey <a class="mention" href="/andreagriffiths11">@andreagriffiths11</a>. Any update on this? </p>

**Comment by 1337hero** [comment]
It's weird that all these years later and this is still a limitation. The silliest of limitations.

**Reply by airtower-luna** [reply]
The limitation doesn't exist any more, you can set up your own publishing workflow: https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#publishing-with-a-custom-github-actions-workflow

**Reply by 1337hero** [reply]
Thanks so much. It was there in front of me the whole time! I figured it out and got it working. Thank you.

**Comment by vladyslavnes** [comment, marked as answer]
As per [this reply](https://github.com/orgs/community/discussions/21853#discussioncomment-4162048)  by @airtower-luna, you can set up your own publishing workflow.

https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#publishing-with-a-custom-github-actions-workflow

**Reply by fahdi** [reply]
Thank you, it was pretty straight forward.

**Comment by johndeighan** [comment]
More complex is bad..., Less complex is good. I won't set up a "workflow" until it's absolutely necessary. GitHub does allow you to publish from a folder named /docs, so you could simply rename your 'public' folder to 'docs', but it's still a pretty silly limitation (personally, I would use another word). As I recall, GitHub's interface makes it look like you could enter a custom name, but it doesn't work. I also agree that since the problem was originally reported in 2018, it's pretty silly (there's that word again) that the problem hasn't been fixed.

Of course, I'm grateful to the GitHub folks for allowing free publishing of static web sites ;-)
